### PR TITLE
pubsub: add gcppubsub/example_test.go

### PR DESCRIPTION
**Pull request by ijt**
This makes progress on #869.

### CODE REVIEW

File: pubsub/gcppubsub/example_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	_, err = s.Receive(ctx)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// Error messages are expected because we don't expect a project named gcppubsub-example-project to exist.
+
+	// Output:
+	// rpc error: code = NotFound desc = Requested project not found or user does not have access to it (project=gcppubsub-example-project). Make sure to specify the unique project identifier and not the Google Cloud Console display name.
```
Comment (vangent) [inline pubsub/gcppubsub/example_test.go]: Matching the exact error string won't work; see the test failures on Travis. Just print something like "This failure is expected...".

File: pubsub/gcppubsub/example_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+	"gocloud.dev/gcp"
+	"gocloud.dev/pubsub"
+	"gocloud.dev/pubsub/gcppubsub"
+	"golang.org/x/oauth2/google"
+)
+
+func Example() {
+	ctx := context.Background()
+	scope := "https://www.googleapis.com/auth/cloud-platform"
```
Comment (vangent) [inline pubsub/gcppubsub/example_test.go]: Can you add some comments to all of this setup stuff? I suspect it's the most useful part of the example.

File: pubsub/gcppubsub/example_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	"fmt"
+	"log"
+
+	"gocloud.dev/gcp"
+	"gocloud.dev/pubsub"
+	"gocloud.dev/pubsub/gcppubsub"
+	"golang.org/x/oauth2/google"
+)
+
+func Example() {
```
Comment (vangent) [inline pubsub/gcppubsub/example_test.go]: Consider breaking it up into publisher and subscriber examples. There will be some duplication but that's fine.

File: pubsub/gcppubsub/example_test.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	}
+
+	_, err = s.Receive(ctx)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// Error messages are expected because we don't expect a project named gcppubsub-example-project to exist.
+
+	// Output:
```
[the remark this answers is not in the document]
Comment (ijt) [reply pubsub/gcppubsub/example_test.go]: Done.

File: pubsub/gcppubsub/example_test.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	t := gcppubsub.OpenTopic(ctx, pubClient, proj, "example-topic", nil)
+
+	// Create a *pubsub.Subscription
+	s := gcppubsub.OpenSubscription(ctx, subClient, proj, "example-subscription", nil)
+
+	if err := t.Send(ctx, &pubsub.Message{Body: []byte("example message")}); err != nil {
+		fmt.Println(err)
+	}
+
+	_, err = s.Receive(ctx)
```
[the remark this answers is not in the document]
Comment (ijt) [reply pubsub/gcppubsub/example_test.go]: Done.

File: pubsub/gcppubsub/example_test.go (hunk 3 above)
Comment (ijt) [reply pubsub/gcppubsub/example_test.go]: Done.

File: pubsub/gcppubsub/example_test.go (hunk 1 above)
Comment (ijt) [reply pubsub/gcppubsub/example_test.go]: Removed the output check.

**Comment by ijt** [comment]
PTAL

**Comment by ijt** [comment]
Sure.

File: pubsub/gcppubsub/example_test.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+	}
+	defer subClient.Close()
+	s := gcppubsub.OpenSubscription(ctx, subClient, proj, "example-subscription", nil)
+	m, err := s.Receive(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("%s\n", m.Body)
+	m.Ack()
+}
```
Comment (jba) [inline pubsub/gcppubsub/example_test.go]: I think you should call s.Shutdown. If someone copy-pastes this into a main program, then I believe the ack might not get sent.

Comment (ijt) [reply pubsub/gcppubsub/example_test.go]: Done.

File: samples/gcmsg/inject_gcp.go [hunk 7]
```diff
@@ -26,21 +26,9 @@ import (
 )
 
 // openGCPSubscription returns the GCP topic based on the subscription ID.
```
Comment (vangent) [inline samples/gcmsg/inject_gcp.go]: This change is unrelated...?

File: pubsub/gcppubsub/example_test.go [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Open a gRPC connection to the GCP Pub Sub API.
+	conn, cleanup, err := gcppubsub.Dial(ctx, creds.TokenSource)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer cleanup()
+	proj := gcp.ProjectID("gcppubsub-example-project")
```
Comment (vangent) [inline pubsub/gcppubsub/example_test.go]: Consider using `gcp.DefaultProjectID(creds)` instead, since that's more likely what a user will do.

File: pubsub/gcppubsub/example_test.go [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+	}
+
+	// Output:
+	// Message send failure is expected due to the fake project name.
+}
+
+func ExampleOpenSubscription() {
+	ctx := context.Background()
+	scope := "https://www.googleapis.com/auth/cloud-platform"
+	creds, err := google.FindDefaultCredentials(ctx, scope)
```
Comment (vangent) [inline pubsub/gcppubsub/example_test.go]: Replace with the FakeCreds?

File: pubsub/gcppubsub/example_test.go (hunk 8 above)
Comment (ijt) [reply pubsub/gcppubsub/example_test.go]: Done.

File: pubsub/gcppubsub/example_test.go (hunk 9 above)
Comment (ijt) [reply pubsub/gcppubsub/example_test.go]: Done.

File: samples/gcmsg/inject_gcp.go (hunk 7 above)
Comment (ijt) [reply samples/gcmsg/inject_gcp.go]: Reverted it.

**Comment by ijt** [comment]
PTAL.

File: pubsub/gcppubsub/example_test.go [hunk 10]
```diff
@@ -0,0 +1,107 @@
+// Copyright 2018 The Go Cloud Authors
```
Comment (vangent) [inline pubsub/gcppubsub/example_test.go]: Please update this to Go Cloud Development Kit Authors.

Comment (ijt) [reply pubsub/gcppubsub/example_test.go]: Done.
